contractrequester/contractrequester.go

```diff
@@ -169,98 +156,40 @@ func (cr *ContractRequester) CallMethod(ctx context.Context, base insolar.Messag
 	if async {
 		return res, nil
 	}
+
 	ctx, cancel := context.WithTimeout(ctx, time.Duration(configuration.NewAPIRunner().Timeout)*time.Second)
 	defer cancel()
-	inslogger.FromContext(ctx).Debug("Waiting for Method results ref=", r.Request)
 
-	var result *reply.CallMethod
+	inslogger.FromContext(ctx).Debug("Waiting for Method results ref=", r.Request)
 
 	select {
 	case ret := <-ch:
 		inslogger.FromContext(ctx).Debug("Got Method results")
 		if ret.Error != "" {
 			return nil, errors.Wrap(errors.New(ret.Error), "CallMethod returns error")
 		}
-		retReply, ok := ret.Reply.(*reply.CallMethod)
-		if !ok {
-			return nil, errors.New("Reply is not CallMethod")
-		}
-		result = &reply.CallMethod{
-			Request: r.Request,
-			Result:  retReply.Result,
-		}
+		return ret.Reply, nil
 	case <-ctx.Done():
 		cr.ResultMutex.Lock()
 		delete(cr.ResultMap, seq)
 		cr.ResultMutex.Unlock()
 		return nil, errors.New("canceled")
 	}
-
-	return result, nil
 }
 
-func (cr *ContractRequester) CallConstructor(ctx context.Context, base insolar.Message, async bool,
-	prototype *insolar.Reference, to *insolar.Reference, method string,
-	argsIn insolar.Arguments, saveAs int) (*insolar.Reference, error) {
-	baseMessage, ok := base.(*message.BaseLogicMessage)
-	if !ok {
-		return nil, errors.New("Wrong type for BaseMessage")
-	}
-
-	msg := &message.CallConstructor{
-		BaseLogicMessage: *baseMessage,
-		PrototypeRef:     *prototype,
-		ParentRef:        *to,
-		Method:           method,
-		Arguments:        argsIn,
-		SaveAs:           message.SaveAs(saveAs),
-	}
-
-	var seq uint64
-	var ch chan *message.ReturnResults
-	if !async {
-		cr.ResultMutex.Lock()
-
-		cr.Sequence++
-		seq = cr.Sequence
-		msg.Sequence = seq
-		ch = make(chan *message.ReturnResults, 1)
-		cr.ResultMap[seq] = ch
-
-		cr.ResultMutex.Unlock()
-	}
+func (cr *ContractRequester) CallMethod(ctx context.Context, inMsg insolar.Message) (insolar.Reply, error) {
+	return cr.Call(ctx, inMsg)
```

Comment:
you check type of returned message in CallConstructor but here you don't, it's strange. What if we get reply.CallConstructor ?

Reply:
also, Call may return other types of replies and pretty much valid, for example RegisterReply